README.md

```diff
@@ -7,3 +7,522 @@ This repository contains submodules of all the repositories used for skywire dep
 * [skywire-utilities](https://github.com/skycoin/skywire-utilities)
 * [skywire-services](https://github.com/skycoin/skywire-services)
 * [skycoin-service-discovery](https://github.com/skycoin/skycoin-service-discovery)
+
+
+## Code Checks & Tests
+
+Dependencies: [go](https://go.dev/) [goimports](https://pkg.go.dev/golang.org/x/tools/cmd/goimports) & [goimports-reviser](https://github.com/incu6us/goimports-reviser)
+
+For any given submodule repository, included is a Makefile with directives such as `format` and `check`.
+`make format check` is currently used by the CI to check pull requests.
+
+`make format check` executes a variation of the following commands
+```
+go mod tidy -v
+[[ -d pkg ]] && goimports -w -local github.com/skycoin/$(basename $(pwd)) ./pkg
+[[ -d cmd ]] && goimports -w -local github.com/skycoin/$(basename $(pwd)) ./cmd
+[[ -d internal ]] && goimports -w -local github.com/skycoin/$(basename $(pwd)) ./internal
+find . -type f -name '*.go' -not -path "./.git/*" -not -path "./vendor/*"  -exec goimports-reviser -project-name github.com/skycoin/$(basename $(pwd)) {} \;
+go clean -testcache &>/dev/null
+[[ -d internal ]] && go test  -cover -timeout=5m -mod=vendor ./internal/...
+[[ -d pkg ]] && go test -cover -timeout=5m -mod=vendor ./pkg/...
+```
+
+The flags used for `go test` may vary from repo to repo
+
+## Building
+
+Dependencies: [go](https://go.dev/)
+
+For any given submodule repository (except skywire-utilities) included is a Makefile with a `build` directive
+
+`make build` executes `go build` for all the binaries that can be built from .go files in `./cmd` subdirectories, for example
+
+```
+	go build ./cmd/skywire-visor
+```
+
+Note that some binaries may have gcc / libc6 dependency unless statically compiled with musl.
+
+## Runtime Dependencies by Service
+
+### Redis
+* address-resolver
+* network-monitor
+* dmsg-discovery
+
+### Postgres
+* transport-discovery
+* route-finder
+* service-discovery
+
+#### DB Setup
+
+Some services use postgresql.
+
+Run postgresql, make a database with UTF-8 character-set
+
+Pass the database, user, and password as envs
+
+```
+export PG_USER=username
+export PG_PASSWORD=pass
+export PG_DATABASE=sampledb
+```
+
+Specify the postgres host and port via flags
+
+```
+route-finder --pg-host localhost --pg-port 5432
+```
+
+All tables created automatically.
+
+
+## Required Services
+
+* address-resolver
+* network-monitor
+* dmsg-discovery
+* dmsg-server
+* transport-discovery
+* route-finder
+* service-discovery
+* setup-node
+
+A list of endpoints corresponding to some of these services in the current deployment
+
+* sd.skycoin.com/api/services?type=proxy
```

Comment:
Not sure if that is any help to users. What are they going to do with these domains?

Reply:
Substitute their domain / subdomain and check those endpoints when they have it set up themselves. It's just for an example or insight.